x-pack/plugins/enterprise_search/public/applications/app_search/components/documents/search_experience/search_experience_content.test.tsx

```diff
@@ -0,0 +1,170 @@
+    myRole: { canManageEngineDocuments: true },
+  };
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    setMockValues(values);
+    setMockSearchContextState(searchState);
+  });
+
+  it('renders', () => {
+    const wrapper = shallow(<SearchExperienceContent />);
+    expect(wrapper.isEmptyRender()).toBe(false);
+  });
+
+  it('passes engineName to the result view', () => {
+    const props = {
+      result: {
+        foo: {
+          raw: 'bar',
+        },
+      },
+    };
+
+    const wrapper = shallow(<SearchExperienceContent />);
+    const resultView: any = wrapper.find(Results).prop('resultView');
+    expect(resultView(props)).toEqual(<ResultView engineName="engine1" {...props} />);
+  });
+
+  it('renders pagination', () => {
+    const wrapper = shallow(<SearchExperienceContent />);
+    expect(wrapper.find(Pagination).exists()).toBe(true);
+  });
+
+  it('renders empty if a search was not performed yet', () => {
+    setMockSearchContextState({
+      ...searchState,
+      wasSearched: false,
+    });
+    const wrapper = shallow(<SearchExperienceContent />);
+    expect(wrapper.isEmptyRender()).toBe(true);
+  });
+
+  it('renders results if a search was performed and there are more than 0 totalResults', () => {
+    setMockSearchContextState({
+      ...searchState,
+      wasSearched: true,
+      totalResults: 10,
+    });
+    const wrapper = shallow(<SearchExperienceContent />);
+    expect(wrapper.find('[data-test-subj="documentsSearchResults"]').length).toBe(1);
+  });
+
+  it('renders a no results message if a non-empty search was performed and there are no results', () => {
+    setMockSearchContextState({
+      ...searchState,
+      resultSearchTerm: 'searchTerm',
+      wasSearched: true,
+      totalResults: 0,
+    });
+    const wrapper = shallow(<SearchExperienceContent />);
+    expect(wrapper.find('[data-test-subj="documentsSearchResults"]').length).toBe(0);
+    expect(wrapper.find('[data-test-subj="documentsSearchNoResults"]').length).toBe(1);
+  });
+
+  describe('when an empty search was performed and there are no results, meaning there are no documents indexed', () => {
+    beforeEach(() => {
+      setMockSearchContextState({
+        ...searchState,
+        resultSearchTerm: '',
+        wasSearched: true,
+        totalResults: 0,
+      });
+    });
+
+    it('renders a no documents message', () => {
+      const wrapper = shallow(<SearchExperienceContent />);
+      expect(wrapper.find('[data-test-subj="documentsSearchResults"]').length).toBe(0);
+      expect(wrapper.find('[data-test-subj="documentsSearchNoDocuments"]').length).toBe(1);
+    });
+
+    it('will include a button to index new documents', () => {
+      const wrapper = mount(<SearchExperienceContent />);
+      expect(
+        wrapper
+          .find(
+            '[data-test-subj="documentsSearchNoDocuments"] [data-test-subj="IndexDocumentsButton"]'
+          )
+          .exists()
+      ).toBe(true);
```

Comment:
[not a blocker, just a suggestion since it looks like there's an i18n error to fix in any case, but feel free to ignore]

This nesting style on this is pretty wild 😬 it might be nice to simplify this with separate assertions. I also sorta like `toHaveLength` over `exists()).toBe(true)` because prettier is less likely to try and insanely indent it, but YMMV

```suggestion
      expect(wrapper.find('[data-test-subj="documentsSearchNoDocuments"]')).toHaveLength(1);
      expect(wrapper.find('[data-test-subj="IndexDocumentsButton"]')).toHaveLength(1);
```
same styling suggested for the other 2 tests in this block, but again, if it's not your preference you can ignore

Reply:
I used a nested selector to be more specific, in case there were any other IndexDocumentsButtons on the page. I want to know in this test asserts that the button is inside of the `documentsSearchNoDocuments` element.

It also just so happens that when I pass the data-test-subj to the EuiButton and do a full mount that it repeats the attribute on 3 separate elements, so I can't actually assert a length of 1, so it's a little cleaner this way then asserting that the length is 3.

I personally prefer exists as it reads cleaner and I don't pay attention to how prettier formats things 😬

Reply:
Ahh gotcha gotcha!!! No worries at all then, this looks good to me! 👍